kex

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# kex is a utility to quickly exec into a kubernetes pod.
+
+[[ -n $DEBUG ]] && set -x
+
+set -eou pipefail
+IFS=$'\n\t'
+
+SCRIPT_DIR="$(dirname "$( readlink -f "${0}" 2>/dev/null || \
+  python -c "import os,sys; print(os.path.realpath(sys.argv[1]))" "${0}" )")"
+
+if [[ -f "${SCRIPT_DIR}/utils.bash" ]]; then
+  source "${SCRIPT_DIR}/utils.bash"
+else
+  source "${SCRIPT_DIR}/../include/utils.bash"
+fi
+
+KEX="${HOME}/.kube/kex"
+
+usage() {
+  cat <<"EOF"
+USAGE:
+  kex             : exec in to the first listed pod
+  kex -ls,--list     : list the available pods
+  kex <NUM>       : exec into a specified pod
+  kex -h,--help   : show this message
+EOF
+  exit 1
+}
+
+POD_NUM=${POD_NUM:-1}
+
+pod_select(){
+  kubectl get pods -o=jsonpath='{range .items[*].metadata.name}{@}{"\n"}{end}' | sed -n ${POD_NUM}p
+}
+
+kube_exec_sh() {
+  kubectl exec -it $(pod_select) sh
+}
+
+kube_exec_bash() {
+  kubectl exec -it $(pod_select) bash
+}
+
+main() {
+  if [[ "$#" -eq 0 ]]; then
+    kube_def_bash
+  elif [[ "$#" -gt 1 ]]; then
+    echo "error: too many flags" >&2
```

Comment:
@lypht Actually I have another comment. Another possible cleanup optimization (for clarity, maintainability etc) could be to split `main()` into `validate_args()` and `exec()` functions. If validation passes, proceed to exec.